### ISSUE TITLE
Unable to build with gym when having multiple targets

### ISSUE THREAD

**Issue by tobihagemann** [fastlane/fastlane#10398]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm unable to build my project with `gym` anymore. I can't tell if it's something new to Xcode 9 or a new fastlane version. I think the last time it worked was with fastlane 2.55.0.

I have a project with two app extension targets. The bundle identifiers are defined with `<redacted>.$(PRODUCT_NAME:rfc1034identifier:lower)`. I'm getting this error when trying to build with `gym`:

```
[15:16:50]: Couldn't automatically detect the provisioning profile mapping
[15:16:50]: There were multiple profiles for bundle identifier(s): <redacted>.$(PRODUCT_NAME:rfc1034identifier:lower)
[15:16:50]: You need to provide an explicit mapping of what provisioning
[15:16:50]: profile to use for each bundle identifier of your app
```

I guess, the issue is that `$(PRODUCT_NAME:rfc1034identifier:lower)` isn't resolved. One app extension should resolve to `<redacted>.saveaction` and the other other one to `<redacted>.documentpicker`.

Btw, everything worked just some weeks ago. I've fixed this by hard-coding the product names instead of using `$(PRODUCT_NAME:rfc1034identifier:lower)` for now. So the issue is kind of resolved but it's still a bug, I suppose?

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.3.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.13.5 (Apple Git-94)           |
| Installation Source         | ~/.rvm/gems/ruby-2.3.4/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.3.4/lib                |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.0                                         |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version "2.53.1"

default_platform :ios

platform :ios do
  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Upload screenshots to iTunes Connect"
  lane :upload_screenshots do
    deliver(skip_binary_upload: true, skip_metadata: true)
  end

  desc "Update metadata in iTunes Connect"
  lane :update_metadata do
    deliver(skip_binary_upload: true, skip_screenshots: true)
  end

  desc "Sync certificates and provisioning profiles for all targets"
  lane :certificates do
    match(type: "appstore")
    match(type: "development")
  end

  desc "Submit a new beta build to TestFlight"
  lane :beta do
    gym(scheme: "<redacted>")
    changelog = File.read("changelog.txt")
    pilot(changelog: changelog)
  end

  desc "Deploy a new version to the App Store"
  desc "This skips uploading of screenshots"
  lane :app_store do
    gym(scheme: "<redacted>")
    deliver(skip_screenshots: true)
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "<redacted>"
apple_id "<redacted>"

team_id "<redacted>"

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.58.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.15.1       |
| io-console                | 0.4.5        |
| slack-notifier            | 1.5.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| plist                     | 3.3.0        |
| CFPropertyList            | 2.3.5        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.57.0       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| json                      | 1.8.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.1        |
| faraday_middleware        | 0.11.0.1     |
| uber                      | 0.1.0        |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.0.2        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.13.4       |
| faraday                   | 0.12.1       |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| terminal-table            | 1.8.0        |
| unicode-display_width     | 1.2.1        |
</details>

*generated on:* **2017-09-22**
</details>

**Comment by tobihagemann** [comment]
Just wanted to mention that I've looked into: https://docs.fastlane.tools/codesigning/xcode-project/#xcode-9-and-up

I'm using `match` but I must admit that I had `Automatically manage signing` activated. It always worked in the past so I didn't know it's not the recommended way. At first, I thought my provisioning profiles were corrupt or something, so I also did a `match nuke` but that didn't help either.

**Comment by ken-jen** [comment]
@tobihagemann can you explain what you mean by "hard-coded" the product names?  I'm having the same issue (though not using match) and would like to implement your workaround.  Is this a project setting in XCode?  An entry in the Fastfile?  A .xcconfig file?

**Comment by tobihagemann** [comment]
I've changed the "Product Bundle Identifier" in the project's "Build Settings". I've simply replaced `$(PRODUCT_NAME:rfc1034identifier:lower)` with the actual string that would've been resolved.

E.g., you have a bundle identifier with `foo.bar.$(PRODUCT_NAME:rfc1034identifier:lower)` and your product name of the target is `baz`. So I'm just setting the bundle identifier to `foo.bar.baz` so that it's not dependent on the product name anymore.
